File: ios/Classes/ErrorCode.swift
//
//  ErrorCode.swift
//  zsdk
//
//  Created by Luis on 1/9/20.
//

import UIKit

enum ErrorCode: String, CaseIterable {
    case EXCEPTION = "EXCEPTION"
    case PRINTER_ERROR = "PRINTER_ERROR"
    case UNKNOWN = "UNKNOWN"
    
    public static func getValueOfName(name: String) -> ErrorCode {
        if(name == EXCEPTION.rawValue) {return EXCEPTION}
        if(name == PRINTER_ERROR.rawValue) {return PRINTER_ERROR}
        if(name == UNKNOWN.rawValue) {return UNKNOWN}
        return UNKNOWN
    }
}

File: lib/src/enumerators/error_code.dart
/// Created by luis901101 on 2020-01-09.
enum ErrorCode {
  SUCCESS,
  EXCEPTION,
  PRINTER_ERROR,
  UNKNOWN,
}

extension ErrorCodeUtils on ErrorCode {
  String get name => toString().split('.').last;

  static ErrorCode? valueOf(String name) {
    for (ErrorCode value in ErrorCode.values)
      if (value.name == name) return value;
    return ErrorCode.UNKNOWN;
  }
}
